maas-schemas/schemas/core/customer.json

```diff
@@ -2,6 +2,88 @@
   "$id": "http://maasglobal.com/core/customer.json",
   "description": "MaaS customer schema",
   "type": "object",
+  "definitions": {
+    "wmpAmount": {
+      "description": "An amount of in-app credits",
+      "type": "integer",
+      "minimum": 0,
+      "examples": [1234]
+    },
+    "wmpBalance": {
+      "description": "An amount of in-app credits, with currency type info",
+      "type": "object",
+      "properties": {
+        "currency": { "const": "WMP" },
+        "amount": { "$ref": "#/definitions/wmpAmount" }
+      },
+      "required": ["currency", "amount"],
+      "additionalProperties": true,
+      "examples": [
+        {
+          "currency": "WMP",
+          "amount": 1234,
+          "type": "charge"
+        }
+      ]
+    },
+    "tokenAmount": {
+      "description": "An amount of in-app tokens",
+      "type": "integer",
+      "minimum": 0,
+      "examples": [1]
+    },
+    "tokenBalance": {
+      "description": "An amount of in-app tokens, with currency type info",
+      "type": "object",
+      "description": "key would typically match tokenId",
+      "properties": {
+        "currency": { "const": "TOKEN" },
+        "tokenId": {
+          "$ref": "http://maasglobal.com/core/components/fare.json#/definitions/tokenId"
+        },
+        "amount": {
+          "oneOf": [{ "$ref": "#/definitions/tokenAmount" }, { "type": "null" }]
```

Comment:
it's safe to say amount cannot be null

Reply:
Actually, I wonder how this has ever worked. The old schema said `"type": ["integer", "null"],` This means an array of length two where the first element is an integer and the second element is `null`.

Reply:
well, maybe schema is wrong so ajv just skip validation 🤷‍♂